dali/pipeline/data/types.h

```diff
@@ -320,7 +325,8 @@ DLL_PUBLIC inline bool IsValidType(const TypeInfo &type) {
     DALI_TYPENAME_REGISTERER(TypeString);                           \
   template <> DLL_PUBLIC DALIDataType TypeTable::GetTypeID<Type>()  \
     DALI_TYPEID_REGISTERER(Type, dtype);                            \
-  DALI_STATIC_TYPE_MAPPING(Type, dtype);
+  DALI_STATIC_TYPE_MAPPING(Type, dtype);                            \
+  DALI_REGISTER_TYPE_IMPL(Type, TypeString, dtype);
```

Comment:
Just a nitpick, it's more about calling the `GetTypeID`, so maybe something named with `invoke` or something related?